### PR TITLE
Prettier Errors

### PR DESCRIPTION
**Pull request by tjvr**
Invalid syntax: line 1: 'baa'

Also, attach `line` and `col` to the `Error` we throw—(is this useful?)

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -420,9 +422,20 @@ describe('errors', () => {
...
+
+  test('have lots of useful attributes', () => {
+    let lexer = compile({ digit: /[0-9]+/ })
+    lexer.reset('40invalid')
+    expect(lexer.next()).toMatchObject({ value: '40' })
+    try {
+      lexer.next()
+    } catch (err) {
+      expect(err).toMatchObject({ line: 1, col: 3 })
+    }
```
Comment (nathan) [inline test/test.js]: Use `expect(…).toThrow`, since this will pass if `lexer.next()` doesn't throw

**Comment by nathan** [comment]
Confusing if you have a syntax error at the end of a line, since it would say the line was blank.

File: test/test.js (hunk 1 above)
Comment (tjvr) [inline test/test.js]: Yeah. But I couldn't get toThrow() to behave like matchObject :(

**Comment by tjvr** [comment]
> Yes, because I like to generate error messages that look like this

Right. But you could already do that by defining an error rule.

**Comment by nathan** [comment]
Fair enough. Probably not, then.

**Comment by nanalan** [comment]
Please add `line` and `col` anyway?
<sup>nearley still doesn't provide the offset even though it does so in err.message</sup>

**Comment by tjvr** [comment]
@nanalan Why? As explained above, you can define an error token.

**Comment by nanalan** [comment]
Actually, that's a good point 😜  Error tokens _do_ look like the way to go.
I was going off of my gut "nearley doesn't provide `line` and `col`" reaction...

**Comment by tjvr** [comment]
> Confusing if you have a syntax error at the end of a line, since it would say the line was blank.

What do you propose? Special-case that -- `Invalid end of line`?

**Comment by nathan** [comment]
I'd be inclined to say just extract the whole line, scanning backward if possible.

**Comment by tjvr** [comment]
@nathan Done!

File: moo.js [hunk 2]
```diff
@@ -322,13 +322,25 @@
...
+        var eol = text.indexOf('\n')
+        if (eol === -1) eol = text.length
+        var line = buffer.slice(start, index + eol)
+        var message = ""
+        message += "invalid syntax at line " + this.line + " col " + this.col + ":\n\n"
+        message += "  " + line + "\n"
+        message += "  "
+        for (var i=this.col - 1; i--; ) {
+          message += " "
+        }
```
Comment (nathan) [inline moo.js]: I think this fails when `this.col > index + 1`, which could happen if you use `save()`. (Also, `Array(this.col).join(' ')` is a nice one-line way to do this if you like that sort of thing.)

Comment (tjvr) [inline moo.js]: I think you're right. Also, sweet! :-)

**Comment by tjvr** [comment]
(Ignore the failed check—that should be fixed now.)

**Pull request merged**